packages/flutter_tools/test/linux/linux_doctor_test.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/doctor.dart';
+import 'package:flutter_tools/src/linux/linux_doctor.dart';
+import 'package:mockito/mockito.dart';
+import 'package:process/process.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+import '../src/mocks.dart';
+
+void main() {
+  group(LinuxDoctorValidator, () {
+    ProcessManager processManager;
+    LinuxDoctorValidator linuxDoctorValidator;
+
+    setUp(() {
+      processManager = MockProcessManager();
+      linuxDoctorValidator = LinuxDoctorValidator();
+    });
+
+    testUsingContext('Returns full validation when clang++ and make are availibe', () async {
+      when(processManager.run(<String>['clang++', '--version'])).thenAnswer((_) async {
+        return FakeProcessResult(
+          stdout: 'clang version 4.0.1-10 (tags/RELEASE_401/final)\junk',
+          exitCode: 0,
+        );
+      });
+      when(processManager.run(<String>[
+        'make',
+        '--version',
+      ])).thenAnswer((_) async {
+        return FakeProcessResult(
+          stdout: 'GNU Make 4.1\njunk',
+          exitCode: 0,
+        );
+      });
+
+      final ValidationResult result = await linuxDoctorValidator.validate();
+      expect(result.type, ValidationType.installed);
+      expect(result.messages, <ValidationMessage>[
+        ValidationMessage('clang++ 4.0.1'),
+        ValidationMessage('GNU Make 4.1'),
+      ]);
+    }, overrides: <Type, Generator>{
+      ProcessManager: () => processManager,
+    });
+
```

Comment:
Please add a test that having a version of clang that's below the minimum gives the expected (partial) result ;)